common/plugins/eu.esdihumboldt.hale.common.headless/src/eu/esdihumboldt/hale/common/headless/transform/filter/InstanceFilterDefinition.java

```diff
@@ -0,0 +1,202 @@
+
+	// private final Set<String> excludeTypes;
+
+	/**
+	 * Default constructor
+	 */
+	public InstanceFilterDefinition() {
+		this.typeFilters = ArrayListMultimap.create();
+		this.unconditionalFilters = new ArrayList<>();
+		this.excludedTypes = new HashSet<String>();
+	}
+
+	/**
+	 * To add type filters
+	 * 
+	 * @param types types of schema
+	 * @param filter {@link Filter}
+	 */
+	public void addTypeFilter(String types, Filter filter) {
+		this.typeFilters.put(types, filter);
+	}
+
+	/**
+	 * To add type filters based on expression
+	 * 
+	 * @param types types of schema
+	 * @param expression filter expression
+	 */
+	public void addTypeFilter(String types, String expression) {
+		this.typeFilters.put(types, createFilter(expression));
+	}
+
+	/**
+	 * To add simple filters
+	 * 
+	 * @param filter {@link Filter}
+	 */
+	public void addUnconditionalFilter(Filter filter) {
+		this.unconditionalFilters.add(filter);
+	}
+
+	/**
+	 * To add simple filters
+	 * 
+	 * @param expression filter expression
+	 */
+	public void addUnconditionalFilter(String expression) {
+		this.unconditionalFilters.add(createFilter(expression));
+	}
+
+	/**
+	 * to get typeFilters
+	 * 
+	 * @return multimap of {@link String} and {@link Filter}
+	 */
+	public Multimap<String, Filter> getTypeFilters() {
+		return this.typeFilters;
+	}
+
+	/**
+	 * to get Unconditional Filters
+	 * 
+	 * @return List of {@link Filter}
+	 */
+	public List<Filter> getUnconditionalFilters() {
+		return this.unconditionalFilters;
+	}
+
+	/**
+	 * Add excluded types from source
+	 * 
+	 * @param value type of Source
+	 */
+	public void addExcludedType(String value) {
+		this.excludedTypes.add(value);
+	}
+
+	/**
+	 * Create a filter from the given test.
+	 * 
+	 * @param expression the filter expression
+	 * @return the filter or may be <code>null</code>
+	 */
+	private Filter createFilter(String expression) {
+		Filter filter = FilterDefinitionManager.getInstance().parse(expression);
+		if (filter == null) // if parsing failed
+			filter = FilterDefinitionManager.getInstance().from("CQL", expression);
+		return filter;
+	}
+
+	/**
+	 * @see eu.esdihumboldt.hale.common.instance.model.Filter#match(eu.esdihumboldt.hale.common.instance.model.Instance)
+	 */
+	@Override
+	public boolean match(Instance instance) {
+
+		return applyFilters(instance);
+	}
+
+	// Filtration using unconditional filters and Typed filters. Instance should
+	// match any of the filter if supplied.
+	private boolean applyFilters(Instance instance) {
+
+		if (this.unconditionalFilters.size() == 0 && this.typeFilters.size() == 0
+				&& this.excludedTypes.size() == 0)
+			return true;
+
+		// applying all filters one by one (also handling lazy evaluation)
+		return (this.unconditionalFilters.size() == 0 ? false : applyUnconditionalFilter(instance))
+				|| (this.typeFilters.size() == 0 ? false : applyTypedFilter(instance))
+				|| (this.excludedTypes.size() == 0 ? false : applyExcludeFilters(instance));
```

Comment:
I don't think there should be an "or" used for the excluded types. As that check should be the least expensive, it should be done first, and false returned on a failed check.